src/helpers/analytics.js

```diff
@@ -0,0 +1,10 @@
+export default function dispatchAnalyticsEvent({type, action, ...data}) {
+  const event = new CustomEvent("component.library.analytics", {
+    detail: {
```

Comment:
Does this need to be nested like this?

Reply:
Are you referring to the formatting or whether our details object needs to be inside an object?

Reply:
Yeah, does it need to be `{ detail: { componentName, action, details } }` or can it just be `{ componentName, action, details }`?

Reply:
From what I can tell, any additional info has to be in the `detail` property...

https://developer.mozilla.org/en-US/docs/Web/API/CustomEvent

For fun, I tried moving the object and it looks it was just disregarded...

![image](https://user-images.githubusercontent.com/62304138/108398141-b0f20580-71e6-11eb-8f1a-011ef4c704d9.png)